File: src/main/java/com/github/xumeng/java/problems/easy/problem0268/Solution0268.java
package com.github.xumeng.java.problems.easy.problem0268;

import java.util.Arrays;

public class Solution0268 {

    public int missingNumber(int[] nums) {
        Arrays.sort(nums);
        if (nums[0] != 0) {
            return 0;
        }
        for (int i = 0; i + 1 < nums.length; i++) {
            if (nums[i + 1] - nums[i] != 1) {
                return nums[i] + 1;
            }
        }
        return nums.length;
    }

}

File: src/main/kotlin/com/github/xumeng/kotlin/problems/easy/problem0268/Solution0268.kt
package com.github.xumeng.kotlin.problems.easy.problem0268

class Solution0268 {

    fun missingNumber(nums: IntArray): Int {
        nums.sort()
        if (nums[0] != 0) {
            return 0
        }
        for (i in 0..<nums.size - 1) {
            if (nums[i + 1] - nums[i] != 1) {
                return nums[i] + 1
            }
        }
        return nums.size
    }

}